pkg/authorization/authconfig.go

```diff
@@ -132,92 +133,103 @@ func (k *annotationAuthTypeDetector) Detect(_ context.Context, res *unstructured
 	return spi.Anonymous, nil
 }
 
-type expressionHostExtractor struct {
-	paths []string
-}
-
-func NewExpressionHostExtractor(paths []string) spi.HostExtractor {
-	return &expressionHostExtractor{paths: paths}
-}
+func UnifiedHostExtractor(extractors ...spi.HostExtractor) spi.HostExtractor { //nolint:gocognit //reason Inlined functions to avoid surface pollution
```

Comment:
What do you mean by "surface pollution"?

Reply:
random minor single use functions all over. "function protected scope" :)

Reply:
Can you rephrase the comment based on ^? That would help me in the future to not get confused :)

Reply:
https://github.com/maistra/odh-platform/pull/49/commits/15e2cadd7c36e1e831278f6667a447c4d23ff9bf